MacOS builds depending on vulkan impl detail (not available)

```
2021-02-04T22:21:21.5739750Z CMake Error at build_tools/cmake/iree_cc_test.cmake:74 (add_executable):
2021-02-04T22:21:21.5740390Z   Target "iree_tools_utils_vm_util_test" links to target
2021-02-04T22:21:21.5741030Z   "iree::hal::vulkan::util::ref_ptr" but the target was not found.  Perhaps a
2021-02-04T22:21:21.5741720Z   find_package() call is missing for an IMPORTED target, or an ALIAS target
2021-02-04T22:21:21.5742240Z   is missing?
2021-02-04T22:21:21.5742620Z Call Stack (most recent call first):
2021-02-04T22:21:21.5743150Z   iree/tools/utils/CMakeLists.txt:38 (iree_cc_test)
```

Any idea why `iree::hal::vulkan::util::ref_ptr` is buried under vulkan utils? (not compiled on MacOS and fails to configure)

--- a/iree/hal/BUILD
+++ b/iree/hal/BUILD
@@ -78,7 +78,6 @@ cc_library(
         "//iree/base:threading",
         "//iree/base:tracing",
         "//iree/base/internal",
-        "//iree/hal/vulkan/util:ref_ptr",
         "@com_google_absl//absl/container:inlined_vector",
         "@com_google_absl//absl/strings",
         "@com_google_absl//absl/types:span",

--- a/iree/hal/CMakeLists.txt
+++ b/iree/hal/CMakeLists.txt
@@ -67,7 +67,6 @@ iree_cc_library(
     iree::base::synchronization
     iree::base::threading
     iree::base::tracing
-    iree::hal::vulkan::util::ref_ptr
   PUBLIC
 )
 

